web/frontend/src/components/slam.vue

```diff
@@ -95,6 +98,66 @@ const fetchSLAMPose = (name: string): Promise<commonApi.Pose> => {
   });
 };
 
+const executeMove = async () => {
+  moveClick = false;
+
+  const req = new motionApi.MoveOnMapRequest();
+
+  /*
+   * set request name
+   * here we set the name of the motion service the user is using
+   */
+  req.setName('builtin');
+
+  const value = await fetchSLAMPose(props.name);
+  // set pose in frame
+  const destination = new commonApi.Pose();
+  destination.setX(Math.abs(destinationMarker.x - value.getX()));
+  destination.setY(Math.abs(destinationMarker.z - value.getY()));
+  destination.setZ(value.getZ());
+  destination.setOX(value.getOX());
+  destination.setOY(value.getOY());
+  destination.setOZ(value.getOZ());
+  destination.setTheta(value.getTheta());
+  req.setDestination(destination);
+
+  // set SLAM resource name
+  const slamResourceName = new commonApi.ResourceName();
+  slamResourceName.setNamespace('rdk');
+  slamResourceName.setType('service');
+  slamResourceName.setSubtype('slam');
+  slamResourceName.setName(props.name);
+  req.setSlamServiceName(slamResourceName);
+
+  // set component name
+  const baseResource = filterResources(props.resources, 'rdk', 'component', 'base');
+  // we are assuming there is only one base that is conducting planning
+  const baseResourceName = new commonApi.ResourceName();
+  baseResourceName.setNamespace('rdk');
+  baseResourceName.setType('component');
+  baseResourceName.setSubtype('base');
+  baseResourceName.setName(baseResource[0]!.name);
+  req.setComponentName(baseResourceName);
+
+  props.client.motionService.moveOnMap(
+    req,
+    new grpc.Metadata(),
+    (error, response) => {
+      if (error) {
+        toast.error(`Error moving: ${error}`);
+        return;
+      }
+      toast.success(`MoveOnMap success: ${response!.getSuccess()}`);
+    }
+  );
+
+};
+
+const executeStopMove = () => {
+  moveClick = true;
+  // TODO: finish
```

Comment:
What is the content of finish?

Reply:
need to figure out how to do this (without adding a method to api)

`executeStopMove` would cancel any existing and executed/executing `motion.MoveOnMap` call.

So if a rover were moving, then a click on stop would have the rover break immediately.